components/ai_chat/content/browser/ai_chat_brave_search_throttle.cc

```diff
@@ -0,0 +1,159 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/content/browser/ai_chat_brave_search_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/check.h"
+#include "base/functional/bind.h"
+#include "brave/components/ai_chat/content/browser/ai_chat_tab_helper.h"
+#include "brave/components/ai_chat/content/browser/page_content_fetcher.h"
+#include "brave/components/ai_chat/core/browser/conversation_handler.h"
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/ai_chat/core/common/utils.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/permission_controller.h"
+#include "content/public/browser/permission_request_description.h"
+#include "content/public/browser/web_contents.h"
+
+namespace ai_chat {
+
+// static
+std::unique_ptr<AIChatBraveSearchThrottle>
+AIChatBraveSearchThrottle::MaybeCreateThrottleFor(
+    std::unique_ptr<Delegate> delegate,
+    content::NavigationHandle* navigation_handle,
+    AIChatService* ai_chat_service) {
+  auto* web_contents = navigation_handle->GetWebContents();
+  if (!web_contents) {
+    return nullptr;
+  }
+
+  if (!delegate || !ai_chat_service ||
+      !ai_chat::IsAIChatEnabled(
+          user_prefs::UserPrefs::Get(web_contents->GetBrowserContext())) ||
```

Comment:
@fallaciousreasoning actually this goes against chromium design principles and PrefService should be passed as a param to `MaybeCreateThrottleFor`

Reply:
fixed in 6cacbb06b4c7421bce8287978c0cd206d069dc0c